File: Part8/Python/FunctionConcepts/FunctionConcepts_2.py
# An example of function variable scope
GlobalVariable = 15
GlobalToBeShadowed = 5

def myFunction():
    global GlobalVariable
    myVariable = 55
    GlobalVariable = 42
    globalToBeShadowed = 15

def main():
    print ( f"Global variable: {GlobalVariable}" )
    print ( f"Global shadowed: {GlobalToBeShadowed}" )
    myFunction()
    print ( f"Function variable: {myVariable}" )  # Will cause NameError
    print ( f"Global variable: {GlobalVariable}" )
    print ( f"Global shadowed: {GlobalToBeShadowed}" )

if __name__ == "__main__":
    main()

File: Part8/Go/FunctionConcepts/FunctionConcepts_2.go
// An example of function variable scope
package main

import "fmt"

var GlobalVariable = 15
var GlobalToBeShadowed = 5

func myFunction() {
    myVariable := 55
    GlobalVariable = 42
    GlobalToBeShadowed := 15
    //_ = myVariable
    //_ = GlobalToBeShadowed
}

func main() {
    fmt.Printf ( "Global variable: %d\n", GlobalVariable )
    fmt.Printf ( "Global shadowed: %d\n", GlobalToBeShadowed )
    myFunction()
    fmt.Printf ( "Function variable: %d\n", myVariable ) // Will cause compile error
    fmt.Printf ( "Global variable: %d\n", GlobalVariable )
    fmt.Printf ( "Global shadowed: %d\n", GlobalToBeShadowed )
}
